### PR TITLE
Add revive to golanci

### PR DESCRIPTION
**Pull request by bshewale**
This PR replace deprecated go-linter using
revive (https://golangci-lint.run/usage/linters/#revive).

We will add "ginkgolinter" in follow up patch once we merged add pre-commit job in openshift/releases.

If we add "ginkgolinter" in this patch then it's causes below issue [1]

```
level=info msg="[config_reader] Config search paths: [./ /home/prow/go/src/github.com/openstack-k8s-operators/dataplane-operator /home/prow/go/src/github.com/openstack-k8s-operators /home/prow/go/src/github.com /home/prow/go/src /home/prow/go /home/prow /home /]"
level=info msg="[config_reader] Used config file .golangci.yaml"
level=error msg="Running error: unknown linters: 'ginkgolinter', run 'golangci-lint help linters' to see the list of supported linters"
```

This patch also Remove fmt dependency from make tidy as per [2]:

The fmt dependency cannot be run if the go.mod file needs an update:
```
 $ make tidy
 go fmt ./...
 go: updates to go.mod needed; to update it:
        go mod tidy
 make: *** [Makefile:103: fmt] Error 1
```

I have also made some changes as per the comment given here [3]

[1]: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_dataplane-operator/147/pull-ci-openstack-k8s-operators-dataplane-operator-main-golangci/1643555465537261568/artifacts/test/build-log.txt 
[2]: https://github.com/openstack-k8s-operators/placement-operator/pull/153
[3]: https://github.com/openstack-k8s-operators/cinder-operator/pull/147#discussion_r1158504341

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -105,11 +105,9 @@ vet: gowork ## Run go vet against code.
...
 
 .PHONY: tidy
-tidy: fmt
-	go mod tidy; \
-	pushd "$(LOCALBIN)/../api"; \
-	go mod tidy; \
-	popd;
+tidy: ## Run go mod tidy on every mod file in the repo
+	go mod tidy
+	cd ./api && go mod tidy
```
[the remark this answers is not in the document]
Comment (bshewale) [reply Makefile:110]: @abays thanks for the review. Yeah it's safe, we will be back into the root path once this : "cd ./api %% go mod tidy" line executes. I got comment here https://github.com/openstack-k8s-operators/cinder-operator/pull/147#discussion_r1158504341 and according i have changed that in all the repos.

**Review by abays — APPROVED** [review]
/lgtm

File: controllers/keystoneapi_controller.go [hunk 2]
```diff
@@ -332,10 +332,10 @@ func (r *KeystoneAPIReconciler) reconcileInit(
 	// expose the service (create service, route and return the created endpoint URLs)
 	//
 	var keystonePorts = map[endpoint.Endpoint]endpoint.Data{
-		endpoint.EndpointPublic: endpoint.Data{
+		endpoint.EndpointPublic: {
```
Comment (marios) [inline controllers/keystoneapi_controller.go:335]: sanity check on this one 
it doesn't seem related to linting

Comment (abays) [reply controllers/keystoneapi_controller.go:335]: It is, because in Golang is it is unnecessary to explicitly declare the type of a map value in the context of assigning a value to a key, since the type of the value is already implied by the map declaration itself.

Comment (marios) [reply controllers/keystoneapi_controller.go:335]: thank you for checking
